version_generated.h Makefile race condition with -j8

While building auth 3.4.1, the homebrew build bot managed to get this error:

```
In file included from secpoll-auth.cc:6:
./version_generated.h:1:2: error: unterminated conditional directive
#ifndef VERSION_GENERATED_H
 ^
```

suggesting that file got truncated somehow. On a retry, the same error popped up but on a different OSX version, confirming the suspicion that this is a race condition.

--- a/pdns/Makefile-recursor
+++ b/pdns/Makefile-recursor
@@ -75,13 +75,19 @@ build_host=$(shell id -u -n)@$(shell hostname -f)
 
 .PHONY: version_generated.h
 version_generated.h:
-	echo '#ifndef VERSION_GENERATED_H' > $@
-	echo '#define VERSION_GENERATED_H' >> $@
-	echo '#include "config.h"' >> $@
-	echo '#define PDNS_VERSION VERSION' >> $@
-	echo '#define BUILD_DATE "$(build_date)"' >> $@
-	echo '#define BUILD_HOST "$(build_host)"' >> $@
-	echo '#endif //!VERSION_GENERATED_H' >> $@
+	@echo "$$create_version_generated_h" > $@
+
+define create_version_generated_h
+#ifndef VERSION_GENERATED_H
+#define VERSION_GENERATED_H
+#include "config.h"
+#define PDNS_VERSION VERSION
+#define BUILD_DATE "$(build_date)"
+#define BUILD_HOST "$(build_host)"
+#endif //!VERSION_GENERATED_H
+endef
+
+export create_version_generated_h
 
 message: 
 	@echo

--- a/pdns/Makefile.am
+++ b/pdns/Makefile.am
@@ -976,10 +976,16 @@ pubsuffix.cc: $(srcdir)/effective_tld_names.dat
 
 .PHONY: version_generated.h
 version_generated.h:
-	@echo '#ifndef VERSION_GENERATED_H' > $@
-	@echo '#define VERSION_GENERATED_H' >> $@
-	@echo '#define PDNS_VERSION "$(pdns_version)"' >> $@
-	@echo '#define DIST_HOST "$(dist_host)"' >> $@
-	@echo '#define BUILD_DATE "$(build_date)"' >> $@
-	@echo '#define BUILD_HOST "$(build_host)"' >> $@
-	@echo '#endif //!VERSION_GENERATED_H' >> $@
+	@echo "$$create_version_generated_h" > $@
+
+define create_version_generated_h
+#ifndef VERSION_GENERATED_H
+#define VERSION_GENERATED_H
+#define PDNS_VERSION "$(pdns_version)"
+#define DIST_HOST "$(dist_host)"
+#define BUILD_DATE "$(build_date)"
+#define BUILD_HOST "$(build_host)"
+#endif //!VERSION_GENERATED_H
+endef
+
+export create_version_generated_h

